misc/python/materialize/cli/run.py

```diff
@@ -157,6 +150,10 @@ def main() -> int:
     )
     args = parser.parse_intermixed_args()
 
+    # `--build-and-reset-only` implies `--reset`
+    if args.build_and_reset_only:
+        args.reset = args.build_and_reset_only
```

Comment:
The way `--build-only`, `--reset`, and `--build-and-reset-only` interact is now quite confusing.

I think you could instead make specifying the existing options (`--build-only --reset`) have the behavior you want/need for RustRover. Specifically: just move the `if args.build_only: return build_retcode` clauses to after the reset action happens. That will preserve the spirit of the existing options, have clear semantics, and IIUC give you the semantics you want for the RustRover config.